src/ui/component/expandable/view.jsx

```diff
@@ -11,48 +12,38 @@ type Props = {
   children: React$Node | Array<React$Node>,
 };
 
-type State = {
-  expanded: boolean,
-};
-
-export default class Expandable extends PureComponent<Props, State> {
-  constructor() {
-    super();
+export default function Expandable(props: Props) {
+  const [expanded, setExpanded] = useState(false);
+  const { children } = props;
+  const ref = useRef();
+  const rect = useRect(ref);
 
-    this.state = {
-      expanded: false,
-    };
-
-    (this: any).handleClick = this.handleClick.bind(this);
+  function handleClick() {
+    setExpanded(!expanded);
   }
 
-  handleClick() {
-    this.setState({
-      expanded: !this.state.expanded,
-    });
-  }
-
-  render() {
-    const { children } = this.props;
-    const { expanded } = this.state;
-
-    return (
-      <div className="expandable">
-        <div
-          className={classnames({
-            'expandable--open': expanded,
-            'expandable--closed': !expanded,
-          })}
-        >
-          {children}
+  return (
+    <div ref={ref}>
+      {rect && rect.height > 120 ? (
```

Comment:
This 120 should be a constant, I think. I'm not sure how I'd use that constant for the styles too, though. Base size is 12px and expandable min height is 10rem (=120).

Reply:
I just moved it up to the top. I think as long as it's reasonable it's fine. In the browser (and possibly desktop sometime) users can scale the font-size so anything exact falls apart. 120 seems fine.